config/defaults/mysql_schema_setup.sql

```diff
@@ -140,12 +140,22 @@ CREATE TABLE `User` (
 /*!50003 SET sql_mode              = 'ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION' */ ;
 DELIMITER ;;
 CREATE DEFINER=`root`@`localhost` PROCEDURE `file_create`(IN username varchar(25), IN filename varchar(50), IN relativePath varchar(2083), IN projectID bigint(20))
-  BEGIN
-    INSERT INTO `File`
-    (Creator, RelativePath, ProjectID, Filename)
-    VALUES (username, relativePath, projectID, filename);
-    SELECT LAST_INSERT_ID();
-  END ;;
+BEGIN
+  IF ( NOT EXISTS ( SELECT `File`.`FileID`
+          FROM `File`
+          WHERE `File`.`ProjectID` =  projectID AND `File`.`RelativePath` = relativePath AND `File`.`Filename` = filename ) ) THEN
```

Comment:
note: this will probably be slow(ish) b/c we don't have an index on these 3, but maintaining the index on this would be a MASSIVE performance and space hog.